src/scaffoldmaker/utils/shieldmesh.py

```diff
@@ -606,3 +664,1178 @@ def generateElements(self, fieldmodule, coordinates, startElementIdentifier, mes
                         meshGroup.addElement(element)
 
         return elementIdentifier
+
+
+class ShieldMesh3D:
+    """
+        Generates a 3D shield mesh.
+    """
+
+    def __init__(self, elementsCountAcross, elementsCountRim, shieldMode=ShieldShape3D.SHIELD_SHAPE_OCTANT_PPP, box_derivatives=None):
```

Comment:
My preference is for the shield mesh to be unaware of any 'modes' limiting ranges, but instead always represents a full sphere limited to ranges of elements in each direction, and eventually in the shell direction as well.
Either this is specified in the constructor, or it's an argument to methods which create nodes and elements.
We'll have to discuss how to do this offline.